src/Transports/MassTransit.RabbitMqTransport/RabbitMqTransport/RabbitMqBasicConsumer.cs

```diff
@@ -160,14 +160,13 @@ public void HandleModelShutdown(object model, ShutdownEventArgs reason)
                 var context = new RabbitMqReceiveContext(exchange, routingKey, _consumerTag, deliveryTag, bodyBytes, redelivered, properties,
                     _context, _receiveSettings, _model, _model.ConnectionContext);
 
-                var receiveLock = _receiveSettings.NoAck ? default : new RabbitMqReceiveLockContext(_model, deliveryTag);
-
                 if (_limit != null)
                     await _limit.WaitAsync(context.CancellationToken).ConfigureAwait(false);
 
                 try
                 {
-                    await Dispatch(deliveryTag, context, receiveLock).ConfigureAwait(false);
+                    await Dispatch(deliveryTag, context, _ => _receiveSettings.NoAck ? default : new RabbitMqReceiveLockContext(_model, deliveryTag))
```

Comment:
Quick thought, is this the _only_ place where we are passing a null lock context? If so, we could simply have a NoLockContext (global instance shared) and avoid the if(!=null) check in the dispatcher.